### PR TITLE
GET boards by userid (Imeth)

### PR DESCRIPTION
**Pull request by Imeth97**
very quick one - dashboard is nowhere near done yet!

future - will make this get teamId by user id and then fetch boards by team id

### CODE REVIEW

File: frontend/src/hooks/useCreateNewBoard.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import useUser from './useUser';
+import { useCallback, useEffect, useState } from 'react';
```
Comment (mursal-ashraf) [inline frontend/src/hooks/useCreateNewBoard.ts:2]: unused import

File: frontend/src/hooks/useBoardOverview.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import useUser from './useUser';
+import { useCallback, useEffect, useState } from 'react';
```
Comment (mursal-ashraf) [inline frontend/src/hooks/useBoardOverview.ts:2]: Was the `supabase-query` library not sufficient for this?

File: frontend/src/components/Dashboard/DashboardUtils.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  Button,
+  DialogContentText,
+  LinearProgress,
+  TextField,
+} from '@mui/material';
+import AlertModal from '../common/AlertModal';
+import { useFormik } from 'formik';
+import { FormSpacer } from '../common/Utils';
+import UseCreateNewBoard from '@/hooks/useCreateNewBoard';
+import WithLoader from '../common/WithLoader/WithLoader';
```
Comment (mursal-ashraf) [inline frontend/src/components/Dashboard/DashboardUtils.tsx:12]: unused import

File: frontend/src/components/common/AlertModal/AlertModal.tsx [hunk 4]
```diff
@@ -8,30 +8,31 @@ import React from 'react';
```
Comment (mursal-ashraf) [inline frontend/src/components/common/AlertModal/AlertModal.tsx:8]: unused import

File: frontend/src/hooks/useBoardOverview.ts (hunk 2 above)
Comment (mursal-ashraf) [reply frontend/src/hooks/useBoardOverview.ts:2]: It would help us reduce the use of any

**Review by mursal-ashraf — APPROVED** [review]
Approved pending queries

File: frontend/src/hooks/useBoardOverview.ts (hunk 2 above)
Comment (Imeth97) [reply frontend/src/hooks/useBoardOverview.ts:2]: got rid of the any, and nah I couldn't find a way to skipcond using the library, so I could not find a way to run dependent queries with the library (fetch of boards _depends_ on us having fetched the user id)

**Review by mursal-ashraf — APPROVED** [review]

**Pull request merged**